subprojects/parseq/src/test/java/com/linkedin/parseq/TestTasks.java

```diff
@@ -48,6 +48,13 @@
  */
 public class TestTasks extends BaseEngineTest {
 
+  private static class UnloggableException extends Exception {
+    @Override
+    public String toString() {
+      throw new RuntimeException("Cannot log an UnloggableException!");
```

Comment:
Change this to UnloggableException to test the scenario Avik mentioned?

Reply:
Yeah, that would do the trick.

Optional: successfully logging the exception thrown by `toString` vs. falling back to a second `catch` block are technically two different scenarios. We don't distinguish between them in the tests because we don't check what actually gets logged, but they do execute different code paths. I would recommend having two test cases:

- The existing `UnloggableException`.
- A `ReallyUnloggableException` where the `toString` throws a `ReallyUnloggableException`.

Reply:
Sounds good. Thanks for the suggestion.

Reply:
Through testing, Logger.warn("...", ex) for an exception that can throw exception in `toString` will not throw exception for some logger. To be safe, I used finally block and added another test for that.

Reply:
If the logging of the second error doesn't actually throw a third error, then I don't think `ReallyUnloggableException` actually exercises the `catch` block. But, without making the logger an injectable dependency, I don't think we even _can_ exercise the `catch`. Still, I think the additional test case is useful, so we should still leave it in.